.github/workflows/ci-circt-nightly.yml

```diff
@@ -16,12 +16,102 @@ on:
   # Run on PRs that target the
   pull_request:
     branches:
-      - ${{ branch-name }}
+      - ci/ci-circt-nightly
 
 jobs:
-  hello:
-    name: Hello World
+  update-branch:
+    name: Integrate staging branch with default branch
     runs-on: ubuntu-20.04
     steps:
-      - name: Hello
-        run: echo "World"
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: main
+      - run: |
+          # Leave the staging branch unchanged if this is a PR.
+          if [[ ${{ github.event_name }} == 'pull_request' ]]; then
+            exit 0
+          fi
```

Comment:
At first I was confused why you put this `if` here rather than a Github Actions-level `if`. I know see that it's clearly do to https://github.com/orgs/community/discussions/45058, could you please add a comment above `run` that explains that?

Reply:
Yes. I want this to be a successful no-op, not a skip. I'll add a comment.